### PR TITLE
Core/Creature: Refactor vendor item code

### PR DESCRIPTION
**Pull request by roc13x**
**Changes proposed**:

- Fix loading referenced vendor items
- Refactored vendor item storage to reduce differences with TC master

**Issues addressed**: Fixes reference vendors

**Tests performed**: Built and fully tested ingame

**Known issues and TODO list**: None

### CODE REVIEW

File: src/server/game/Events/GameEventMgr.cpp [hunk 1]
```diff
@@ -1597,7 +1601,13 @@ void GameEventMgr::UpdateWorldStates(uint16 event_id, bool Activate)
...
 }
 
-GameEventMgr::GameEventMgr() : isSystemInit(false) { }
+GameEventMgr::GameEventMgr() : isSystemInit(false)
+{
+}
+
+GameEventMgr::~GameEventMgr()
+{
+}
```
Comment (Ovahlord) [inline src/server/game/Events/GameEventMgr.cpp:1611]: why did you move the destructor from header over here when it's empty?

Comment (roc13x) [reply src/server/game/Events/GameEventMgr.cpp:1611]: This PR is largely a cherry-pick of my older PR to TC master. Shauren added to the PR before merge, and that was one of the changes for some reason. Codestyle I guess

**Comment by funjoker** [comment]
Thanks for the PR

**Comment by Ovahlord** [comment]
@roc13x you have a little homework to do: guild vendors should open a empty vendor window when no rewards are available. at the moment the vendor window does not even open
